### PR TITLE
Added Scal Asum expression fusion test

### PR DESCRIPTION
**Pull request by przemektmalon**
- Added a Scal + Asum expression tree fusion test

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -55,4 +55,6 @@ include_directories(
 )
 
 add_subdirectory(${SYCLBLAS_TEST}/unittest)
-#add_subdirectory(${SYCLBLAS_TEST}/exprtest)
+if(ENABLE_EXPRESSION_TESTS)
```
Comment (mehdi-goli) [inline test/CMakeLists.txt]: Please use option and set the default value for ENABLE_EXPRESSION_TESTS

File: test/unittest/CMakeLists.txt [hunk 2]
```diff
@@ -28,7 +28,7 @@ project(syclblas_unittest)
 
 set(SYCLBLAS_UNITTEST ${CMAKE_CURRENT_SOURCE_DIR})
 
-include_directories(${SYCLBLAS_TEST} ${BLAS_INCLUDE_DIRS})
+include_directories(${SYCLBLAS_TEST})
```
Comment (mehdi-goli) [inline test/unittest/CMakeLists.txt]: Is this related to this PR?

File: test/CMakeLists.txt (hunk 1 above)
Comment (przemektmalon) [reply test/CMakeLists.txt]: Added to main CMakeLists

File: test/unittest/CMakeLists.txt (hunk 2 above)
Comment (przemektmalon) [reply test/unittest/CMakeLists.txt]: Not strictly. Just noticed that this include was not necessary, and that ${BLAS_INCLUDE_DIRS} was always empty for me, so I removed it. Should I add it back ?

Comment (mehdi-goli) [reply test/unittest/CMakeLists.txt]: yes, it is used when you install the system blas in a custom place
